svm/src/transaction_processing_result.rs

```diff
@@ -14,14 +19,23 @@ pub trait TransactionProcessingResultExtensions {
     fn flattened_result(&self) -> TransactionResult<()>;
 }
 
+pub enum ProcessedTransaction {
+    /// Transaction was executed, but if execution failed, all account state changes
+    /// will be rolled back except deducted fees and any advanced nonces
+    Executed(Box<ExecutedTransaction>),
+    /// Transaction was not able to be executed but fees are able to be
+    /// collected and any nonces are advanceable
+    FeesOnly(Box<FeesOnlyTransaction>),
+}
```

Comment:
Outside the scope of this PR, but we may want to re-evaluate this at some later point.
Doing a box-allocation per transaction is likely bad for performance - though I'm sure there are **many** such sins in our current pipeline.